components/join-us-modal/form/create-fields.js

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import validator from "../../../js/form-validator";
+
+module.exports = function(user, profile) {
+  let step1 = {
+    custom_name: {
+      type: `text`,
+      placeholder: ``,
+      defaultValue: profile.custom_name || user.name,
+      fieldClassname: `form-control`,
+      validator: [
+        validator.emptyValueValidator(),
+        validator.maxLengthValidator(70)
+      ]
+    },
+    email: {
+      type: `text`,
+      defaultValue: user.email,
+      fieldClassname: `form-control`
+    },
+    newsletter: {
+      // TODO:FIXME: not sure if Pulse API supports this?
```

Comment:
this is supported.

https://github.com/mozilla/network-pulse-api/blob/master/pulseapi/urls.py#L73

Reply:
@Pomax what data do I POST to this route? Is it just `email` along with `csrf_token` and `nonce`?

https://github.com/mozilla/network-pulse-api/blob/065dd3196b9598eb69a1d380e29b7eaf82d92b84/pulseapi/utility/newsletters/signup.py#L56-L83

Reply:
you shouldn't need the `nonce` for this route, and the payload should have an `email` field as well as a `source` field. The `source` field is used to say which URL the signup request came from so usually that's  `window.location.toString()`